spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/ChannelNode.java

```diff
@@ -21,21 +21,18 @@
 /**
  * @author Andy Clement
  * @author David Turanski
+ * @author Ilayaperumal Gopinathan
  */
 public class ChannelNode extends AstNode {
 
-	private ChannelType channelType;
+	private String nameComponent;
 
-	private List<String> nameComponents;
+	private List<String> labelComponents;
```

Comment:
We will need to do some follow-up renaming in general (changing "channel" to "destination" in most cases, for example), so we can include this on the list. In the *typical* case, the "labelComponents" would only be present for a "tap" (since that means the source destination is one fro a stream, thus having `<stream>.<label>` syntax). However, since someone could explicitly create a destination with a dot in it now, it's generally valid. IIUC, the expected pattern is now: `identifier[.identifier]*`, and that entire thing is actually the NAME of the destination, so I'd argue that even "nameComponent" is misnamed currently. We need to decide on the right name for 2 things:

1. the required identifier that either stands alone or precedes the first DOT
2. the list of optional identifiers that are themselves separated by dots, following the first dot

All that said, I feel as though this arbitrary distinction based on the first dot is no longer necessary. Isn't the whole thing just the name? Do we even need to know if it's a "tap" anymore?